src/Microsoft.Azure.WebJobs.Host/Executors/FunctionInstanceTraceWriter.cs

```diff
@@ -1,39 +0,0 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for license information.
-
-using System;
-using System.Diagnostics;
-
-namespace Microsoft.Azure.WebJobs.Host.Executors
-{
-    /// <summary>
-    /// The <see cref="TraceWriter"/> passed to function instances. This class adds function instance details
-    /// to all trace events.
-    /// </summary>
-    internal class FunctionInstanceTraceWriter : TraceWriter
-    {
-        private TraceWriter _innerWriter;
-        private IFunctionInstance _instance;
-        private Guid _hostInstanceId;
-
-        public FunctionInstanceTraceWriter(IFunctionInstance instance, Guid hostInstanceId, TraceWriter innerWriter, TraceLevel level)
-            : base(level)
-        {
-            _innerWriter = innerWriter;
-            _instance = instance;
-            _hostInstanceId = hostInstanceId;
-        }
-
-        public override void Trace(TraceEvent traceEvent)
-        {
-            if (traceEvent == null)
-            {
-                throw new ArgumentNullException("traceEvent");
-            }
-
-            traceEvent.AddFunctionInstanceDetails(_hostInstanceId, _instance.FunctionDescriptor, _instance.Id);
```

Comment:
Where does this happen in your new ILogger pipeline? Also, if we're still supporting a TraceWriter binding, don't we still need this to happen for such traces?

Reply:
These were useful for people plugging in their own TraceWriters to our pipeline. Today, they'd plug in their own ILogger. We wrap all the calls to functions in a logger.BeginScope(<function details>) call, which makes the function details available to any loggers that are registered. This is how the App Insights logger gets the details about a specific message.

The instance details we're providing are slightly different -- but I did notice that I'm missing the hostInstanceId, for example. I'll add that to the scope, and we can continue adding more details as we get requests to do so.